tests/ytt/external_db_test.go

```diff
@@ -0,0 +1,69 @@
+package ytt
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("External DB", func() {
+
+	var ctx RenderingContext
+	var data map[string]string
+	var templates []string
+
+	BeforeEach(func() {
+		templates = []string{
+			pathToFile("config/postgres"),
+			pathToFile("tests/ytt/postgres/postgres-values.yml"),
+		}
+	})
+
+	JustBeforeEach(func() {
+		ctx = NewRenderingContext(templates...).WithData(data)
+	})
+
+	Context("disabled", func() {
+
+		BeforeEach(func() {
+			data = map[string]string{
+				"cf_db.admin_password":   "vz9o2hbkh6x7ztnd6ua6",
+				"capi.database.password": "9v5ljlmxje8uv32gwl5q",
+				"uaa.database.password":  "yevxad9e0pvgc8l6osnt",
+				"capi.database.host":     "",
+				"uaa.database.host":      "",
+			}
+		})
+
+		It("should have cf-db namespace and cf-db-postgresql statefulset", func() {
+
+			Expect(ctx).To(ProduceYAML(
+				And(
+					WithNamespace("cf-db"),
+					WithStatefulSet("cf-db-postgresql").WithNamespace("cf-db")),
+			))
+		})
+	})
+
+	Context("enabled", func() {
+
+		BeforeEach(func() {
+			data = map[string]string{
+				"cf_db.admin_password":   "vz9o2hbkh6x7ztnd6ua6",
+				"capi.database.password": "9v5ljlmxje8uv32gwl5q",
+				"uaa.database.password":  "yevxad9e0pvgc8l6osnt",
+				"capi.database.host":     "foo.bar",
+				"uaa.database.host":      "foo.bar",
+			}
+		})
+
+		It("should not have cf-db* namespace/statefulset", func() {
+
+			Expect(ctx).To(Not(ProduceYAML(
+				And(
+					WithNamespace("cf-db"),
+					WithStatefulSet("cf-db-postgresql").WithNamespace("cf-db")),
+			)))
```

Comment:
Would the "negative" testing interface make more sense if it was something like this?

```
Expect(ctx).To(ProduceYAML(
  And(
    WithoutNamespace("cf-db"),
    WithoutStatefulSet("cf-db-postgres"),
  )
))
```

Reply:
I can see this is more readable/understandable but its at the engineering cost of one matcher (not matcher) versus 2x the constructor functions.

I dont have a strong opinion either way.

Reply:
It's probably more clear, but will increase the interface quite a bit. It would give more granular control - for example:
```
Expect(ctx).To(ProduceYAML(
  And(
    WithNamespace("cf-db"),
    WithStatefulSet("cf-db-postgres").WithoutNamespace("cf-db")),
  )
))
```
It'd be great material for refactor/getting community engagement. We could mark this material as good first issues.

Reply:
To avoid duplicating checks, can we write a matcher that is simply the negation of another matcher? At least that way we don't have to maintain both matchers and avoid them getting out of sync.

Reply:
@jamespollard8 and I decided to use the standard gomega not matcher in the end.